src/tree.ts

```diff
@@ -325,4 +310,51 @@ export class Tree {
   getProof(input: ProofInput): Proof {
     return createProof(this.rootNode, input);
   }
+
+  private getParentNodes(bitstring: GindexBitstring, expand = false): Node[] {
+    let node = this.rootNode;
+
+    // Keep a list of all parent nodes of node at gindex `index`. Then walk the list
+    // backwards to rebind them "recursively" with the new nodes without using functions
+    const parentNodes: Node[] = [this.rootNode];
+
+    // Ignore the first bit, left right directions are at bits [1,..]
+    // Ignore the last bit, no need to push the target node to the parentNodes array
+    for (let i = 1; i < bitstring.length - 1; i++) {
+      if (node.isLeaf()) {
+        if (!expand) {
+          throw new Error(ERR_INVALID_TREE);
+        } else {
+          node = zeroNode(bitstring.length - i);
+        }
+      }
+
+      // Compare to string directly to prevent unnecessary type conversions
+      if (bitstring[i] === "1") {
+        node = node.right;
+      } else {
+        node = node.left;
+      }
+
+      parentNodes.push(node);
+    }
+
+    return parentNodes;
+  }
+
+  private rebindNodeToRoot(bitstring: GindexBitstring, parentNodes: Node[], newNode: Node): void {
```

Comment:
Can you add a nice JSDoc comment like in setHashObjectFn explaining this function?